### DISCUSSION TITLE
React Testing Library Set Up Example That shows how to wrap Refine

### DISCUSSION THREAD

**Discussion by vinclou** [refinedev/refine#2466] [Ideas]
There's an example of using Refine with e2e cypress setup, but would it be possible to create one for integration testing? Specifically, creating a custom render function with the wrapper that includes Refine's integrated providers. The closest example I found is https://github.com/pankod/refine/blob/next/packages/core/test/index.tsx in the core test-lib. I agree that cypress can cover many areas in the app, but if you want Integration testing, there's barely any info on how to set one up.

I can attempt to make a PR about it if you could guide me through the possible problem space.

**Comment by omeraplak** [comment]
Hey @vinclou 👋,
Thank you for contacting us! We have an example in our ant design and mui packages, but it's a great idea to add it to our document! 🚀

https://github.com/pankod/refine/blob/next/packages/antd/test/index.tsx#L28

**Reply by omeraplak** [reply]
Did this work for you? @vinclou

**Reply by vinclou** [reply]
I haven't get back to jest set up yet; Will do by the end of the week or early next week; Thank you!
